api/core/tools/provider/builtin/agentql/tools/query_data.yaml

```diff
@@ -0,0 +1,78 @@
+identity:
+  name: query_data
+  author: Tiny Fish
+  label:
+    en_US: AgentQL Query Data
+description:
+  human:
+    en_US: Extract structured data from a given URL.
+  llm: A tool for extracting structured data matching a given AgentQL query from a given URL. # Introduction passed to LLM, in order to make LLM better understand this tool, we suggest to write as detailed information about this tool as possible here, so that LLM can understand and use this tool
+parameters:
+  - name: url
+    type: string
+    required: true
+    label:
+      en_US: URL
+    human_description:
+      en_US: The public URL of the webpage to extract data from.
+    form: form
+  - name: query
+    type: string
+    required: true
+    label:
+      en_US: AgentQL Query
+    human_description:
+      en_US: The AgentQL query to execute. Read more at https://docs.agentql.com/agentql-query.
+    form: form
```

Comment:
I noticed that `form` value requires users to provide value manually, but `llm` value requires another block to pass the value in. Is there a way to allow both? And if not, it seems that `llm` is better for both url and query in this case. Am I on the right track?

On the other side, I don't expect LLM to "generate" a query, but it's more like "I need a way to dynamically provide it, not by LLM exactly"